### DISCUSSION TITLE
Glyphs look squished after reinstalling Arch.

### DISCUSSION THREAD

**Discussion by dharmx** [ryanoasis/nerd-fonts#781] [Q&A]
Hello, I just reinstalled Arch yesterday then installed [nerd-fonts-complete](https://aur.archlinux.org/packages/nerd-fonts-complete) with yay and then when I tried load eww, rofi, polybar to overlap as illustrated below.

Is this related to my fontconfg? Also, note that all Nerd Fonts are behaving this way Hack, JetBrains, Iosevka, etc. 

Sorry, I'm a noob when it comes to fonts. Also, if you require any extra information, please let me know!

---

#### After reinstall
![image](https://user-images.githubusercontent.com/80379926/152937129-d4d37907-e84d-426b-a0b6-36aa3d41bb1c.png)

#### Before reinstall
![image](https://user-images.githubusercontent.com/80379926/152937852-660d658f-9be0-44f3-b59d-eb1a741a0bd5.png)

---

#### After reinstall
![lol](https://user-images.githubusercontent.com/80379926/152937198-c55739e0-95b3-475b-ad22-54e11db99012.png)

#### Before Reinstall
![image](https://user-images.githubusercontent.com/80379926/152937685-2ecfb8d4-f28b-446f-ac21-6dfbbfb356bc.png)

### Fontconfig
Placed in `$XDG_CONFIG_HOME/fontconfig/font.conf`

---
```xml
<?xml version="1.0"?>
<!DOCTYPE fontconfig SYSTEM "fonts.dtd">

<fontconfig>
  <!-- Default serif font -->
  <alias binding="strong">
    <family>serif</family>
    <prefer>
      <family>Roboto</family>
    </prefer>
  </alias>

  <!-- Default sans-serif font -->
  <alias binding="strong">
    <family>sans-serif</family>
    <prefer>
      <family>Roboto</family>
    </prefer>
  </alias>

  <!-- Default monospace font -->
  <alias binding="strong">
    <family>monospace</family>
    <prefer>
      <family>JetBrainsMono Nerd Font</family>
    </prefer>
  </alias>

  <!-- Default system-ui font -->
  <alias binding="strong">
    <family>system-ui</family>
    <prefer>
      <family>Roboto Condensed</family>
    </prefer>
  </alias>

  <match target="font">
    <edit name="lcdfilter" mode="assign">
      <const>lcddefault</const>
    </edit>
  </match>
</fontconfig>
```

**Comment by Finii** [comment]
Unfortunately my VM with Arch installed got lost along the way, so I can not try this out directly.

What I see is that neither `EWW` nor `Rofi` depend on `nerd-fonts-complete`. Their UIs have been designed with some _other_ specific font in mind - and it works obviously. Probably it is not a good idea to substitute a font that is used for GUI elements with a different one. Although I personally also do that, at work we have a lot of GUIs that are designed with the Microsoft fonts or Google fonts. I have these not installed, and of course sometimes (often) there are glitches in the GUI, widths not matching, texts too long etc. That is to be expected, unfortunately, to some degree always.

In your `font.conf` I see two different 'sets' of changes, for one you use `Roboto` as `system-ui` and more; and you use `JetBrainMono NF` for `monospace`.

I doubt that `EWW` or `Rofi` try to fetch a `monospace` font for the UI elements, so maybe `Roboto`.

There is a `nerd-fonts-roboto-mono` AUR, do you use that? There is no mention of `Roboto Nerd Fonts` in your config.

- Did you raise an Issue with the AUR package? In the first instance that is who should have checked stuff ;-)
- I would remove the `font.conf` and check if problem persists (then it is a font substitution problem)
- If that works check if the `system-ui` font specification is used by `EWW` and `Rofi`, by disabling just that lines
- `Rofi` looks fine in my eyes, just the font is more condensed, but if you exchange a font ... some change is expected?
- `EWW` looks indeed strange, but changing the font for this kind of UI elements strikes me as bad idea anyhow?
- There might be problems with the fonts themselves. The AURs have been updated to use Release 2.1.0. There was an [Issue from Arch](https://bugs.archlinux.org/task/66212), that led to a (in my eyes _breaking_) [change](https://github.com/ryanoasis/nerd-fonts/commit/59c45ba4eff539d2a83a7d7004f056b9860d3253) of font logic in the Nerd Font repo between 2.0.0 and 2.1.0. I'm a bit bewildered on the Arch font setup.
